http/oidc/src/main/java/org/wildfly/security/http/oidc/TokenValidator.java

```diff
@@ -288,6 +291,33 @@ private static String getAccessTokenHash(String accessTokenString, String jwsAlg
 
     }
 
+
+    private static class NonceValidator implements ErrorCodeValidator {
+        private OidcHttpFacade.Cookie cookie;
+
+        public NonceValidator(OidcHttpFacade.Cookie cookie) {
+            this.cookie = cookie;
+        }
+
+        public ErrorCodeValidator.Error validate(JwtContext jwtContext) throws MalformedClaimException {
+            JwtClaims idJwtClaims = jwtContext.getJwtClaims();
+            IDToken idToken = new IDToken(idJwtClaims);
+
+            if (cookie != null) {
+                String sessionIdValue = Oidc.getCryptographicValue(cookie.getValue());
+                String nonceValue = idToken.getNonce();
+                if (nonceValue == null || !sessionIdValue.equals(nonceValue)) {
+                    return new ErrorCodeValidator.Error(INVALID_SESSION_RANDOM_VALUE,
+                            log.invalidNonceValue(nonceValue == null ? "null" : nonceValue));
```

Comment:
I don't think specifying "null" explicitly is needed here.

Reply:
done